Minio/Helper/utils.cs

```diff
@@ -905,5 +906,52 @@ public static DateTime From8601String(string dt)
             return DateTime.Parse(dt, null, System.Globalization.DateTimeStyles.RoundtripKind);
         }
 
+        public static Uri GetBaseUrl(string endpoint)
+        {
+            if (String.IsNullOrEmpty(endpoint))
+            {
+                throw new ArgumentException(String.Format("{0} is the value of the endpoint. It can't be null or empty.", endpoint),"endpoint");
+            }
+            if (endpoint.EndsWith("/"))
+            {
+                endpoint = endpoint.Substring(0, endpoint.Length - 1);
+            }
+            if (!endpoint.StartsWith("http") && !BuilderUtil.IsValidHostnameOrIPAddress(endpoint))
+            {
+                throw new InvalidEndpointException(String.Format("{0} is invalid hostname.", endpoint),"endpoint");
+            }
+            string conn_url;
+            if (endpoint.StartsWith("http"))
```

Comment:
```suggestion
            if (!endpoint.StartsWith("http"))
```
is this what you meant?

Reply:
No. We just need the input in the format of `IP:Port`, `DomainName:Port` or `DomainName.TLD` etc. and then if `WithSSL()` is called, we add the https scheme (or http otherwise) when constructing the URL. Same checks are made in MinioClient.cs, MinioClientBuilder.cs I think.